### ISSUE TITLE
Select Problem When Editing Multiple Carts In RdLibrary

### ISSUE THREAD

**Issue by RichStivers** [ElvishArtisan/rivendell#238]
When editing groups of carts in RdLibrary, selecting carts with <CTRL><left-mouse> only works correctly with carts adjacent to the carts already selected.

When a cart is non-adjacent to the carts already selected and is <CTRL><left-mouse> selected, the previously selected carts are erroneously unselected.

**Comment by deltecent** [comment]
I can confirm this behavior exists in the stable and master branches.

Hopefully someone can explain why selections are handled this way:

```
void RDListView::mouseButtonClickedData(int button,QListViewItem *item,
					const QPoint &pt,int col)
{
  QListViewItem *l;
  bool contiguous;

  if((selectionMode()!=QListView::Extended)||(item==NULL)||(button!=1)) {
    return;
  }

  //
  // Get Selected Range
  //
  l=item;
  contiguous=true;
  while((l=l->itemAbove())!=NULL) {
    if(!l->isSelected()) {
      contiguous=false;
    }
    if(!contiguous) {
      setSelected(l,false);
    }
  }
  l=item;
  contiguous=true;
  while((l=l->itemBelow())!=NULL) {
    if(!l->isSelected()) {
      contiguous=false;
    }
    if(!contiguous) {
      setSelected(l,false);
    }
  }
}
```

**Comment by deltecent** [comment]
Fixed in PR #375
